src/Talis/Persona/Client/Base.php

```diff
@@ -203,6 +214,57 @@ protected function getHTTPClient()
     }
 
     /**
+     * Retrieve the Persona client version
+     * @return string Persona client version
+     */
+    protected function getClientVersion()
+    {
+        $version = $this->getCacheBackend()->fetch(self::COMPOSER_VERSION_CACHE_KEY);
+        if ($version) {
+            return $version;
+        }
+
+        $composerFileContent = file_get_contents(
+            __DIR__. '/../../../../composer.json'
+        );
+
+        if ($composerFileContent === false) {
+            return 'unknown';
+        }
+
+        $composer = json_decode($composerFileContent, true);
+        if (isset($composer['version']) === false) {
+            return 'unknown';
+        }
+
+        $this->getCacheBackend()->save(
+            self::COMPOSER_VERSION_CACHE_KEY,
+            $composer['version'],
+            self::COMPOSER_VERSION_CACHE_TTL_SEC
+        );
+
+        return $composer['version'];
+    }
+
+    /**
+     * Returns a unique id for tracing this request.
+     * If there is already a value set as a header it uses that, otherwise it
+     * generates a new one and sets that on $_SERVER
+     * @return string
+     */
+    protected function getRequestId()
+    {
+        $requestId = null;
+        if (array_key_exists('HTTP_X_REQUEST_ID', $_SERVER)) {
+            $requestId = $_SERVER['HTTP_X_REQUEST_ID'];
```

Comment:
So Now I think that this is wrong (or the rest of the internet is). Everywhere I look on Google it's `X-REQUEST-ID`.